src/Image/Processor.php

```diff
@@ -441,13 +441,20 @@ public function resize($new_dimension, $use_height = false, $squared = false, $r
 
 		$resource = imagecreatetruecolor($new_w, $new_h);
 
-		if ($this->image_type == IMAGETYPE_PNG)
-		{
-			imagealphablending($resource, false);
-			imagesavealpha($resource, true);
-			$transparent = imagecolorallocatealpha($resource, 255, 255, 255, 127);
-			imagefilledrectangle($resource, 0, 0, $new_w, $new_h, $transparent);
-		}
+		$transparencyIndex = imagecolortransparent($this->resource);
+		$transparencyColor = array('red' => 255, 'green' => 255, 'blue' => 255);
+		if ($transparencyIndex >= 0)
+		{
+			$transparencyColor = imagecolorsforindex($this->resource, $transparencyIndex);
+		}
+		$transparencyIndex = imagecolorallocate(
+			$resource,
+			$transparencyColor['red'],
+			$transparencyColor['green'],
+			$transparencyColor['blue']
+		);
+		imagefill($resource, 0, 0, $transparencyIndex);
+		imagecolortransparent($resource, $transparencyIndex);
```

Comment:
This removed the image type check `if ($this->image_type == IMAGETYPE_PNG)`. Any consequences?

Reply:
I don't believe so, as the check wasn't even hitting in the first place as the constants weren't defined with unique values.